### ISSUE TITLE
Installation issue on Scientific Linux 6.6 (RedHat 6.6 clone)

### ISSUE THREAD

**Issue by Alexander-Barth** [bicycle1885/EzXML.jl#102]
When I try to install the lastest version of EzXML (0.9.1) in Julia 1.1.0, I get the following error:
```
  Building EzXML ─────→ `~/.julia/packages/EzXML/r19gO/deps/build.log`
┌ Error: Error building `EzXML`: 
│ [ Info: Downloading https://github.com/bicycle1885/XML2Builder/releases/download/v1.0.2/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz to /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/downloads/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz...
│ ERROR: LoadError: LibraryProduct(nothing, ["libxml2"], :libxml2, "Prefix(/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr)") is not satisfied, cannot generate deps.jl!
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] #write_deps_file#152(::Bool, ::Function, ::String, ::Array{LibraryProduct,1}) at /home/ulg/gher/abarth/.julia/packages/BinaryProvider/4F5Hq/src/Products.jl:414
│  [3] (::getfield(BinaryProvider, Symbol("#kw##write_deps_file")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(write_deps_file), ::String, ::Array{LibraryProduct,1}) at ./none:0
│  [4] top-level scope at none:0
│  [5] include at ./boot.jl:326 [inlined]
│  [6] include_relative(::Module, ::String) at ./loading.jl:1038
│  [7] include(::Module, ::String) at ./sysimg.jl:29
│  [8] include(::String) at ./client.jl:403
│  [9] top-level scope at none:0
│ in expression starting at /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/build.jl:48
[14:42:43] ######################################################################## 100.0%
└ @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1075
```

The issue seems to be related to zlib:
```
Libdl.dlopen("/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so")
ERROR: could not load library "/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so"
/lib64/libz.so.1: version `ZLIB_1.2.3.3' not found (required by /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so)
Stacktrace:
```

The system comes with the zlib version 1.2.3 (zlib-1.2.3-29.el6.src.rpm):

```
$ rpm -ql zlib
/lib64/libz.so.1
/lib64/libz.so.1.2.3
```

Only the patch-level version is different. Is it possible that `libxml2.so` is build in a way that it does not require a specific patch level?

**Comment by bicycle1885** [comment]
I cannot reproduce the problem on our CentOS 7.2.1511 machine. EzXML.jl is supposed to not depend on the system's libz. Can you reproduce the problem even after clean install (e.g., removing all Julia packages located in `~/.julia`)?

```
ksato@elwood000 ~> julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.0 (2019-01-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(v1.1) pkg> add EzXML
   Cloning default registries into `~/.julia`
   Cloning registry from "https://github.com/JuliaRegistries/General.git"
     Added registry `General` to `~/.julia/registries/General`
 Resolving package versions...
 Installed EzXML ────────── v0.9.1
 Installed BinaryProvider ─ v0.5.3
  Updating `~/.julia/environments/v1.1/Project.toml`
  [8f5d6c58] + EzXML v0.9.1
  Updating `~/.julia/environments/v1.1/Manifest.toml`
  [b99e7846] + BinaryProvider v0.5.3
  [8f5d6c58] + EzXML v0.9.1
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [8ba89e20] + Distributed
  [b77e0a4c] + InteractiveUtils
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [6462fe0b] + Sockets
  [8dfed614] + Test
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
  Building EzXML → `~/.julia/packages/EzXML/r19gO/deps/build.log`

(v1.1) pkg> test EzXML
   Testing EzXML
 Resolving package versions...
    Status `/tmp/tmpJZ7WLo/Manifest.toml`
  [b99e7846] BinaryProvider v0.5.3
  [8f5d6c58] EzXML v0.9.1
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [ade2ca70] Dates  [`@stdlib/Dates`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [76f85450] LibGit2  [`@stdlib/LibGit2`]
  [8f399da3] Libdl  [`@stdlib/Libdl`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [44cfe95a] Pkg  [`@stdlib/Pkg`]
  [de0858da] Printf  [`@stdlib/Printf`]
  [3fa0cd96] REPL  [`@stdlib/REPL`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [ea8e919c] SHA  [`@stdlib/SHA`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [8dfed614] Test  [`@stdlib/Test`]
  [cf7118a7] UUIDs  [`@stdlib/UUIDs`]
  [4ec0a83e] Unicode  [`@stdlib/Unicode`]
Test Summary: | Pass  Total
Error         |   88     88
[ Info: the following two warnings are expected:
┌ Warning: XMLError: htmlParseCharRef: missing semicolon from HTML parser (code: 7, line: 2)
└ @ EzXML ~/.julia/packages/EzXML/r19gO/src/error.jl:95
┌ Warning: XMLError: htmlParseCharRef: missing semicolon from HTML parser (code: 7, line: 4)
└ @ EzXML ~/.julia/packages/EzXML/r19gO/src/error.jl:95
Test Summary: | Pass  Total
Reader        |   28     28
Test Summary: | Pass  Total
Writer        |    3      3
[ Info: the following warning is expected:
┌ Warning: caught 4 errors; show the first one
└ @ EzXML ~/.julia/packages/EzXML/r19gO/src/error.jl:79
Test Summary: | Pass  Total
Parser        |   29     29
Test Summary: | Pass  Total
Stream Reader |  224    224
Test Summary: | Pass  Total
Constructors  |   71     71
Test Summary: | Pass  Total
Traversal     |  144    144
Test Summary: | Pass  Total
Properties    |   41     41
Test Summary: | Pass  Total
Construction  |  102    102
Test Summary: | Pass  Total
Validation    |    4      4
Test Summary: | Pass  Total
XPath         |   43     43
Test Summary: | Pass  Total
Misc          |   10     10
Test Summary: | Pass  Total
Examples      |    3      3
   Testing EzXML tests passed

julia> versioninfo()
Julia Version 1.1.0
Commit 80516ca202 (2019-01-21 21:24 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)
```

**Comment by Alexander-Barth** [comment]
Thanks for having a look at this issue. Unfortunately, even with no previously installed package I have this installation problem. For this test I moved my previous `.julia` folder out of the way, but I verified that no packages are installed in Julia:

```
$ mv .julia .julia-bak
$ ~/opt/julia-1.1.0/bin/julia 
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.0 (2019-01-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(v1.1) pkg> status
    Status `/home/users/a/b/abarth/.julia/environments/v1.1/Project.toml`
  (empty environment)

(v1.1) pkg> add EzXML
   Cloning default registries into `~/.julia`
   Cloning registry from "https://github.com/JuliaRegistries/General.git"
     Added registry `General` to `~/.julia/registries/General`
 Resolving package versions...
 Installed EzXML ────────── v0.9.1
 Installed BinaryProvider ─ v0.5.3
  Updating `/home/users/a/b/abarth/.julia/environments/v1.1/Project.toml`
  [8f5d6c58] + EzXML v0.9.1
  Updating `/home/users/a/b/abarth/.julia/environments/v1.1/Manifest.toml`
  [b99e7846] + BinaryProvider v0.5.3
  [8f5d6c58] + EzXML v0.9.1
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
  Building EzXML → `~/.julia/packages/EzXML/r19gO/deps/build.log`
┌ Error: Error building `EzXML`: 
│ [ Info: Downloading https://github.com/bicycle1885/XML2Builder/releases/download/v1.0.2/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz to /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/downloads/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz...
│ ERROR: LoadError: LibraryProduct(nothing, ["libxml2"], :libxml2, "Prefix(/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr)") is not satisfied, cannot generate deps.jl!
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] #write_deps_file#152(::Bool, ::Function, ::String, ::Array{LibraryProduct,1}) at /home/ulg/gher/abarth/.julia/packages/BinaryProvider/4F5Hq/src/Products.jl:414
│  [3] (::getfield(BinaryProvider, Symbol("#kw##write_deps_file")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(write_deps_file), ::String, ::Array{LibraryProduct,1}) at ./none:0
│  [4] top-level scope at none:0
│  [5] include at ./boot.jl:326 [inlined]
│  [6] include_relative(::Module, ::String) at ./loading.jl:1038
│  [7] include(::Module, ::String) at ./sysimg.jl:29
│  [8] include(::String) at ./client.jl:403
│  [9] top-level scope at none:0
│ in expression starting at /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/build.jl:48
[20:30:00] ######################################################################## 100.0%
└ @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1075
```
The file `~/.julia/packages/EzXML/r19gO/deps/build.log` contains exactly the same error message as printed to the screen:
```
[ Info: Downloading https://github.com/bicycle1885/XML2Builder/releases/download/v1.0.2/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz to /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/downloads/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz...
ERROR: LoadError: LibraryProduct(nothing, ["libxml2"], :libxml2, "Prefix(/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr)") is not satisfied, cannot generate deps.jl!
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] #write_deps_file#152(::Bool, ::Function, ::String, ::Array{LibraryProduct,1}) at /home/ulg/gher/abarth/.julia/packages/BinaryProvider/4F5Hq/src/Products.jl:414
 [3] (::getfield(BinaryProvider, Symbol("#kw##write_deps_file")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(write_deps_file), ::String, ::Array{LibraryProduct,1}) at ./none:0
 [4] top-level scope at none:0
 [5] include at ./boot.jl:326 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1038
 [7] include(::Module, ::String) at ./sysimg.jl:29
 [8] include(::String) at ./client.jl:403
 [9] top-level scope at none:0
in expression starting at /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/build.jl:48
[20:30:00] ######################################################################## 100.0%
```

It seems that `libxml2.so` still links to `libz.so.1` and the provided one in RedHat 6.6 (and clones) is too old. 

```
ldd /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so
/home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so: /lib64/libz.so.1: version `ZLIB_1.2.3.3' not found (required by /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so)
	linux-vdso.so.1 =>  (0x00002aaaaaaab000)
	/cm/shared/apps/libibverbs_copy/libibverbs.so (0x00002aaaaae84000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00002aaaab0ae000)
	libz.so.1 => /lib64/libz.so.1 (0x00002aaaab2b2000)
	libm.so.6 => /lib64/libm.so.6 (0x00002aaaab4c8000)
	libc.so.6 => /lib64/libc.so.6 (0x00002aaaab74d000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00002aaaabae1000)
	/lib64/ld-linux-x86-64.so.2 (0x0000003c91800000)
``` 

I am wondering what the output of `ldd .../deps/usr/lib/libxml2.so` be on Centos 7.

There is no libz.so library provided in EzXML (as it was in the past). 

```
$ find /home/ulg/gher/abarth/.julia/packages/EzXML -name "libz*"
# no results  
```

Is `libxml2.so` compiled in a way, that it no longer depends on `libz.so` ?
I see a couple of undefined symbols related to libz in libxml2:

```
$ nm /home/ulg/gher/abarth/.julia/packages/EzXML/r19gO/deps/usr/lib/libxml2.so | grep gz
                 U gzclose
                 U gzdirect@@ZLIB_1.2.2.3
                 U gzdopen
                 U gzopen64@@ZLIB_1.2.3.3
                 U gzread
                 U gzwrite
```

Your help is really appreciated!

```
julia> versioninfo()
Julia Version 1.1.0
Commit 80516ca202 (2019-01-21 21:24 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, sandybridge)
```

**Comment by Alexander-Barth** [comment]
I noticed that the XML2Builder release 1.0.2 does not contains zlib in the folder lib:

https://github.com/bicycle1885/XML2Builder/releases/download/v1.0.2/XML2Builder.v2.9.9.x86_64-linux-gnu.tar.gz

while the initial release 1.0.0 does contain zlib: 

https://github.com/bicycle1885/XML2Builder/releases/download/v1.0.0/XML2Builder.x86_64-linux-gnu.tar.gz

**Comment by bicycle1885** [comment]
Ah, that seems to be a problem. Thank you for diagnosing it! I'll take a look more carefully.

**Comment by PeterNorlindh** [comment]
I'm unable to install/build on Ubuntu 19.04. Is this the same issue? Please let me know you'd if like any further info from me.

```
julia> using EzXML
[ Info: Precompiling EzXML [8f5d6c58-4d21-5cfd-889c-e3ad7ee6a615]
ERROR: LoadError: EzXML.jl is not installed properly, run Pkg.build("EzXML") and restart Julia.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at /home/peter/.julia/packages/EzXML/QlX60/src/EzXML.jl:111
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1044
 [5] include(::Module, ::String) at ./sysimg.jl:29
 [6] top-level scope at none:2
 [7] eval at ./boot.jl:319 [inlined]
 [8] eval(::Expr) at ./client.jl:393
 [9] top-level scope at ./none:3
in expression starting at /home/peter/.julia/packages/EzXML/QlX60/src/EzXML.jl:110
ERROR: Failed to precompile EzXML [8f5d6c58-4d21-5cfd-889c-e3ad7ee6a615] to /home/peter/.julia/compiled/v1.0/EzXML/Hh8vo.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1203
 [3] _require(::Base.PkgId) at ./loading.jl:960
 [4] require(::Base.PkgId) at ./loading.jl:858
 [5] require(::Module, ::Symbol) at ./loading.jl:853

(v1.0) pkg> build EzXML
  Building EzXML → `~/.julia/packages/EzXML/QlX60/deps/build.log`
┌ Error: Error building `EzXML`: 
│ ERROR: LoadError: LibraryProduct(nothing, ["libxml2"], :libxml2, "Prefix(/home/peter/.julia/packages/EzXML/QlX60/deps/usr)") is not satisfied, cannot generate deps.jl!
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] #write_deps_file#152(::Bool, ::Function, ::String, ::Array{LibraryProduct,1}) at /home/peter/.julia/packages/BinaryProvider/TcAwt/src/Products.jl:414
│  [3] (::getfield(BinaryProvider, Symbol("#kw##write_deps_file")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(write_deps_file), ::String, ::Array{LibraryProduct,1}) at ./none:0
│  [4] top-level scope at none:0
│  [5] include at ./boot.jl:317 [inlined]
│  [6] include_relative(::Module, ::String) at ./loading.jl:1044
│  [7] include(::Module, ::String) at ./sysimg.jl:29
│  [8] include(::String) at ./client.jl:392
│  [9] top-level scope at none:0
│ in expression starting at /home/peter/.julia/packages/EzXML/QlX60/deps/build.jl:48
└ @ Pkg.Operations /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1096
```

**Comment by ljschumacher** [comment]
I was able to install on Ubuntu 18.04 but not build on Mac OSX
`julia> Pkg.build("EzXML")
  Building EzXML → `~/.juliapro/JuliaPro_v1.1.1.1/packages/EzXML/CPWLR/deps/build.log`
┌ Error: Error building `EzXML`: 
│ ERROR: LoadError: LibraryProduct(nothing, ["libxml2"], :libxml2, "Prefix(/Users/linus/.juliapro/JuliaPro_v1.1.1.1/packages/EzXML/CPWLR/deps/usr)") is not satisfied, cannot generate deps.jl!
│ Stacktrace:
│  [1] error(::String) at /Applications/JuliaPro-1.1.1.1.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
│  [2] #write_deps_file#156(::Bool, ::Function, ::String, ::Array{LibraryProduct,1}) at /Users/vagrant/worker/juliapro-release-osx1011-0_6/build/tmp_julia/Julia-1.1.app/Contents/Resources/julia/share/julia/stdlib/v1.1/BinaryProvider/src/Products.jl:414
│  [3] (::getfield(BinaryProvider, Symbol("#kw##write_deps_file")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(write_deps_file), ::String, ::Array{LibraryProduct,1}) at ./none:0
│  [4] top-level scope at none:0
│  [5] include_relative(::Module, ::String) at /Applications/JuliaPro-1.1.1.1.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
│  [6] include(::Module, ::String) at /Applications/JuliaPro-1.1.1.1.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
│  [7] include(::String) at ./client.jl:403
│  [8] top-level scope at none:0
│ in expression starting at /Users/linus/.juliapro/JuliaPro_v1.1.1.1/packages/EzXML/CPWLR/deps/build.jl:48
└ @ Pkg.Operations /Users/vagrant/worker/juliapro-release-osx1011-0_6/build/tmp_julia/Julia-1.1.app/Contents/Resources/julia/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1138`

However, homebrew tells me libxml2 is installed and up-to-date

**Comment by ararslan** [comment]
This will be fixed with the next version of BinaryBuilder/BinaryProvider, but that will require Julia 1.3 or later. According to @staticfloat, the more immediate, cross-version fix would be to include the build.jl file generated from ZlibBuilder alongside that from XML2Builder.

**Comment by bicycle1885** [comment]
I've confirmed that the problem has been fixed by #121.
